modules/virtualhub/main.tf

```diff
@@ -0,0 +1,13 @@
+
+resource "azurerm_virtual_hub" "virtual_hub" {
+  for_each = var.virtual_hubs != null ? var.virtual_hubs : {}
```

Comment:
remove for_each here and the looping to be done at the parent module that calls this sub-module

Reply:
This is not possible since we want to capture the state using "moved" block. This is required as per the "moved" syntax

Reply:
May I check why this is not possible as it was possible for other resources like firewall which you have moved into a submodule and used the moved block